providers/task.rb

```diff
@@ -87,12 +102,28 @@
   end
 end
 
+action :end do
+  if @current_resource.exists
+    if @current_resource.status != :running
+      Chef::Log.debug "#{@new_resource} is not running - nothing to do"
+    else
+      cmd =  "schtasks /End /TN \"#{@current_resource.task_name}\" "
+      shell_out!(cmd, {:returns => [0]})
```

Comment:
I'm not 100% sure what end does, but if the task finishes between when you got its status and you run this command, what will happen? I'm guessing you don't get a helpful error code if it fails